pkg/daemon/fake/manager.go

```diff
@@ -123,6 +124,77 @@ func (t *testMgr) DeleteService(name string) error {
 	return t.svcList.remove(name)
 }
 
+func (t *testMgr) DeleteServiceAndDependents(name string) error {
+	existingSvcs, err := t.GetServices()
+	if err != nil {
+		return errors.Wrap(err, "could not determine existing Windows services")
+	}
+	// Nothing to do if it already does not exist
+	if _, present := existingSvcs[name]; !present {
+		return nil
+	}
+
+	winSvcObj, err := t.OpenService(name)
+	if err != nil {
+		return errors.Wrapf(err, "unable to open %s service handler", name)
+	}
+	defer winSvcObj.Close()
+
+	status, err := winSvcObj.Query()
+	if err != nil {
+		return errors.Wrapf(err, "unable to query %s service status", name)
+	}
+	// Ensure no dependent services are running before we can safely stop this one
+	if status.State != svc.Stopped {
+		if err := t.removeDependentServices(name, existingSvcs); err != nil {
+			return err
+		}
+	}
+	klog.Infof("removing service %s", name)
```

Comment:
Is this logging really necessary?
This is only going to make unit tests noisy.
This function can't be used in a real scenario

We should be getting enough feedback from asserts and error messages